examples/example_fit_eos.py

```diff
@@ -0,0 +1,328 @@
+    nul = TH_data[:,0]*0.
+    PTH_covariances = np.array([[nul, nul, nul], [nul, TH_data[:,1], nul], [nul, nul, np.power(TH_data[:,2]*4.184*0.0004, 2.)]]).T
+
+    # Ok, so we've now loaded the new enthalpy data. Let's combine it with the volume data we loaded earlier and create some flags:
+    flags = ['H' for P in PTH_data[:,0]]
+    flags.extend(['V' for P in PTV_data[:,0]])
+
+    PTp_data = np.concatenate([PTH_data, PTV_data])
+    PTp_covariances = np.concatenate([PTH_covariances, PTV_covariances])
+
+    V_mask = [i for i, flag in enumerate(flags) if flag == 'V']
+    
+    # And now we can fit our data!
+    fitted_eos = burnman.tools.fit_PTp_data(mineral = per_opt,
+                                            p_flags = flags,
+                                            fit_params = ['V_0', 'K_0', 'Kprime_0', 'grueneisen_0', 'q_0', 'F_0'],
+                                            data = PTp_data,
+                                            data_covariances = PTp_covariances,
+                                            verbose = False)
+
+
+    
+    temperatures = np.linspace(1., 3000., 101)
+    pressures = np.array([1.e5] * 101)
+    plt.plot(temperatures, per_HP.evaluate(['V'], pressures, temperatures)[0]*1.e6, linestyle='--', label='HP')
+    plt.plot(temperatures, per_SLB.evaluate(['V'], pressures, temperatures)[0]*1.e6, linestyle='--', label='SLB')
+    plt.plot(temperatures, per_opt.evaluate(['V'], pressures, temperatures)[0]*1.e6, label='Optimised fit')
+    
+    DS1997_data = np.loadtxt('../burnman/data/input_fitting/Dubrovinsky_Saxena_1997_TV_periclase.dat')
+    plt.errorbar(DS1997_data[:,0], DS1997_data[:,1], yerr=DS1997_data[:,2], linestyle='None', label='Dubrovinsky and Saxena (1997)')
+    plt.errorbar(H1976_T, H1976_V*1.e6, yerr=H1976_Verr*1.e6,
+                 marker='o', markersize=5, linestyle='None', label='Hazen (1976)')
+
+    plt.legend(loc='upper left')
+    plt.xlim(0., temperatures[-1])
+    plt.xlabel('Temperature (K)')
+    plt.ylabel('Volume (cm^3/mol)')
+    plt.show()
+    
+    # Print the optimized parameters
+    print('Optimized equation of state:')
+    burnman.tools.pretty_print_values(fitted_eos.popt, fitted_eos.pcov, fitted_eos.fit_params)
+    print('')
+    
+    # Create a corner plot of the covariances
+    fig=burnman.nonlinear_fitting.corner_plot(fitted_eos.popt, fitted_eos.pcov, fitted_eos.fit_params)
+    plt.show()
+
+    # Now let's plot the volume residuals
+    fig = plt.figure()
+    ax = fig.add_subplot(1, 1, 1)
+    
+    sd_limit = 3 # plot discrete colors up to this integer limit 
+    cmap = plt.cm.RdYlBu
+    cmap.set_under('k')
+    cmap.set_over('k')
+    bounds = np.linspace(-sd_limit, sd_limit, sd_limit*2+1)
+    norm = colors.BoundaryNorm(bounds, cmap.N)
+    
+    im = ax.scatter(PTV_data[:,0]/1.e9, PTV_data[:,1], c=fitted_eos.weighted_residuals[V_mask], cmap=cmap, norm=norm, s=50)
+    fig.colorbar(im, ax=ax)
```

Comment:
Good catch Sanne! I've tweaked the example to be closer to the neat version in contrib/eos_fitting. All the figures should now have sensible axes.